src/components/cursor-controller.js

```diff
@@ -255,77 +258,71 @@ AFRAME.registerComponent("cursor-controller", {
   },
 
   _handleTouchStart: function(e) {
-    if (!this.isMobile || this.hasPointingDevice) return;
-
-    const touch = e.touches[0];
-    if (touch.clientY / window.innerHeight >= 0.8) return true;
-    this.mousePos.set(touch.clientX / window.innerWidth * 2 - 1, -(touch.clientY / window.innerHeight) * 2 + 1);
-    this._updateRaycasterIntersections();
+    if (!this.isMobile || this.hasPointingDevice || this.activeTouch) return;
 
-    // update cursor position
-    if (!this.isGrabbing) {
-      const intersections = this.el.components.raycaster.intersections;
-      if (intersections.length > 0 && intersections[0].distance <= this.data.maxDistance) {
-        const intersection = intersections[0];
-        this.data.cursor.object3D.position.copy(intersection.point);
-        this.currentDistance = intersections[0].distance;
-        this.currentDistanceMod = 0;
-      } else {
-        this.currentDistance = this.data.maxDistance;
+    for (let i = e.touches.length - 1; i >= 0; i--) {
+      const touch = e.touches[i];
+      if (touch.clientY / window.innerHeight < 0.8) {
+        this.activeTouch = touch;
+        break;
       }
     }
+    if (!this.activeTouch) return;
 
-    this._setLookControlsEnabled(false);
-
-    // Set timeout because if I don't, the duck moves is picked up at the
-    // the wrong offset from the cursor: If the cursor started below and
-    // to the left, the duck lifts above and to the right by the same amount.
-    // I don't understand exactly why this is, since I am setting the
-    // cursor object's position manually in this function, but something else
-    // must happen before cursor-grab ends up doing the right thing.
-    // TODO : Figure this out.
-    window.setTimeout(() => {
-      this.data.cursor.emit("cursor-grab", {});
-    }, 40);
-
-    this.lastTouch = touch;
-  },
-
-  _updateRaycasterIntersections: function() {
-    const raycaster = this.el.components.raycaster.raycaster;
+    // Update the ray and cursor positions
+    const raycasterComp = this.el.components.raycaster;
+    const raycaster = raycasterComp.raycaster;
     const camera = this.data.camera.components.camera.camera;
+    const cursor = this.data.cursor;
+    this.mousePos.set(
+      this.activeTouch.clientX / window.innerWidth * 2 - 1,
```

Comment:
just curious: is `-1` always the correct offset?

Reply:
I'm not 100% sure if `clientX` and `innerWidth` are the correct values here, but if they are then the `-1` is the correct offset.
I'm counting on `clientX / innerWidth` to return a value between [0,1]. Then I multiply by 2 to spread it over [0,2]. Then I subtract 1 to put it into the normalized device coordinates that threejs expects ([-1, 1]).

Reply:
Perhaps `touch.screenX` or `touch.pageX` or `window.outerWidth` is more correct, I'm not sure.